libsolidity/analysis/ControlFlowRevertPruner.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolidity/analysis/ControlFlowRevertPruner.h>
+
+#include <libsolutil/Algorithms.h>
+
+namespace solidity::frontend
+{
+
+void ControlFlowRevertPruner::run()
+{
+	for (auto& [pair, flow]: m_cfg.allFunctionFlows())
```

Comment:
Can you update the comments here? The first just builds the lookup table m_calls, the second does the actual work on the graph. Nothing is pending.